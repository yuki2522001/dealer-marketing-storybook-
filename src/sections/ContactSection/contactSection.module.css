.contact {
  display: flex;
  padding: 0 90px;
  justify-content: space-between;
  margin: 0 auto;
  max-width: 1140px;
}

.contact-description {
  font-size: var(--f-regular);
  color: var(--text-secondary);
  margin: 30px 0 10px;
}

.contact-submit {
  display: flex;
  width: 650px;
  height: 40px;
  justify-content: space-between;
}

.contact-input {
  display: flex;
  align-items: center;
  margin-left: 20px;
  width: 370px;
  border-bottom: 1px solid var(--text-secondary);
}

.contact-input > i {
  font-size: var(--f-primary);
  color: var(--text-secondary);
  margin-right: 10px;
}

.contact-input:hover {
  border-bottom: 2px solid var(--text-secondary);
}

.contact-dash-first,
.contact-dash-secondary {
  background-color: var(--border-primary);
}

.contact-dash-first {
  height: 20px;
  margin-bottom: 40px;
}

.contact-dash-secondary {
  height: 100px;
}
