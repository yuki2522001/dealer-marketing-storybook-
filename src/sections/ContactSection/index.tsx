import React, { memo } from "react";
import { IconType } from "@components/common/Icon";
import { ButtonType } from "@components/common/Button";
import { Text, Input, Button, Icon } from "@components/common";
import { SizeType } from "@components/common/Text";
import styleContact from "./contactSection.module.css";

const ContactSection: React.FC = () => (
  <>
    <div
      data-testid="contact-section"
      className={styleContact["contact-dash-first"]}></div>
    <div className={styleContact.contact}>
      <div className={styleContact["contact-info"]}>
        <Text size={SizeType.medium} text="Get Tips & Tricks every Week!" />
        <div className={styleContact["contact-description"]}>
          <Text text="Join our newsletter and get news in your inbox every week! We hate spam too, so no worries about this." />
        </div>
      </div>
      <div className={styleContact["contact-submit"]}>
        <div className={styleContact["contact-input"]}>
          <Icon iconName={IconType.letter} />
          <Input type="text" placeholder="Your email..." />
        </div>
        <Button type={ButtonType.secondary} text="Subscribe" />
      </div>
    </div>
    <div className={styleContact["contact-dash-secondary"]}></div>
  </>
);

export default memo(ContactSection);
