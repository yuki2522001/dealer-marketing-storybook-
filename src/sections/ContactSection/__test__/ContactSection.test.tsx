import { render } from "@testing-library/react";
import ContactSection from "..";

describe("ContactSection component", () => {
  test("Should render ContactSection component", () => {
    const { getByTestId } = render(<ContactSection />);
    const contactSection = getByTestId("contact-section");
    expect(contactSection).toBeInTheDocument();
  });

  test("Matches snapshot", () => {
    const { container } = render(<ContactSection />);
    expect(container).toMatchSnapshot();
  });
});
