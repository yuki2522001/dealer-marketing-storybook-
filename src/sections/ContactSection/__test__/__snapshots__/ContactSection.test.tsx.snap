// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactSection component Matches snapshot 1`] = `
<div>
  <div
    data-testid="contact-section"
  />
  <div>
    <div>
      <p
        data-testid="text"
      >
        Get Tips & Tricks every Week!
      </p>
      <div>
        <p
          data-testid="text"
        >
          Join our newsletter and get news in your inbox every week! We hate spam too, so no worries about this.
        </p>
      </div>
    </div>
    <div>
      <div>
        <i
          data-testid="icon"
        />
        <input
          data-testid="input-value"
          placeholder="Your email..."
          type="text"
        />
      </div>
      <button
        data-testid="button"
      >
        Subscribe
      </button>
    </div>
  </div>
  <div />
</div>
`;
